src/metabase/driver/postgres.clj

```diff
@@ -298,6 +302,73 @@
   [driver [_ arg]]
   (sql.qp/->honeysql driver [:percentile arg 0.5]))
 
+(defn- datetime-diff-helper [x y unit]
+  (case unit
+    (:year :day)
+    (hx/cast
+     :integer
+     (hsql/call
+      :date_part
+      (hx/literal unit)
+      (hsql/call
+       (case unit :year :age :day :-)
+       (hsql/call :date_trunc (hsql/raw "'day'") y)
+       (hsql/call :date_trunc (hsql/raw "'day'") x))))
+
+    :week
+    (hx// (datetime-diff-helper x y :day) 7)
+
+    :month
+    (hx/cast
+     :integer
+     (hx/+
+      (hx/* 12 (datetime-diff-helper x y :year))
+      (hsql/call
+       :date_part
+       (hsql/raw "'month'")
+       (hsql/call
+        :age
+        (hsql/call :date_trunc (hsql/raw "'day'") y)
+        (hsql/call :date_trunc (hsql/raw "'day'") x)))))
+
+    (:hour :minute :second)
+    (let [ex            (hsql/call :extract :epoch (hx/->timestamptz x))
+          ey            (hsql/call :extract :epoch (hx/->timestamptz y))
+          positive-diff (fn [a b]
+                          (hx/cast
+                           :integer
+                           (hx/floor
+                            (if (= unit :second)
+                              (hx/- b a)
+                              (hx// (hx/- b a) (case unit :hour 3600 :minute 60))))))]
+      (hsql/call :case (hsql/call :<= ex ey) (positive-diff ex ey) :else (hx/* -1 (positive-diff ey ex))))
+
+    (throw (ex-info (tru "Invalid datetime-diff unit: {0}" unit)
+                    {:valid-units [:year :month :week :day :hour :minute :second]
+                     :bad-unit unit}))))
```

Comment:
this is missing `qp.error-type/` info -- add a `:type` key with `qp.error-type/invalid-query` or something

Reply:
Updated in https://github.com/metabase/metabase/commit/8b90edd4e33f7c4990af9e5b5baaf3a0550db851